tests/e2e/default/enketo/pregnancy-complete-a-delivery.wdio-spec.js

```diff
@@ -247,7 +247,7 @@ describe('Contact Delivery Form', () => {
       source_id: '',
       user: {
         contact_id: offlineUser.contact._id,
-        name: offlineUser.username,
+        name: offlineUser.contact.name,
```

Comment:
This change actually demonstrates this fix in action!  The regression scenario [from the forum](https://forum.communityhealthtoolkit.org/t/current-user-data-hydration-with-db-object-issue-using-cht4/3880/4) was doing the exact same thing as the `delivery` form where they are trying to load the user's contact document directly into the `inputs/user` group.  The complication is that the `inputs/user` group _already contains data from the users `user-settings` doc._  So, when you use the contact selector to load the user's contact into the same group, it will override the data that is already there.  

In this case, both the user's contact and the `user-settings` document contain a `name` field.  For the `user-settings` it is the username. For the contact, it is just the normal contact name value.  Because the contact-selector functionality in non-relevant `inputs` was broken before this PR, the user's contact data was not actually being loaded and so the `name` was just populated with the username from the `user-settings` document.  Now that this PR has fixed the contact selector functionality, we see `inputs/user/name` getting overridden with the value from the user's contact (which is the proper behavior that we expect).